modules/build/src/main/scala/scala/build/preprocessing/directives/UsingMainClassDirectiveHandler.scala

```diff
@@ -0,0 +1,51 @@
+package scala.build.preprocessing.directives
+import scala.build.Position
+import scala.build.errors.BuildException
+import scala.build.options.BuildOptions
+import scala.build.preprocessing.ScopePath
+
+case object UsingMainClassDirectiveHandler extends UsingDirectiveHandler {
+
+  def name        = "Main class"
+  def description = "Specify default main class"
+
+  def usage = "using main-class _main class_ | using mainClass _main class_"
+
+  override def usageMd =
+    """`using main-class `_main class_
+      |
+      |`using mainClass `_main class_""".stripMargin
+
+  override def examples = Seq(
+    "using main-class helloWorld"
+  )
+
+  override def handle(
+    directive: Directive,
+    cwd: ScopePath
+  ): Option[Either[BuildException, BuildOptions]] =
+    directive.values match {
+      case Seq("main-class", mainClass) =>
+        val options = BuildOptions(
+          mainClass = Some(mainClass)
+        )
+        Some(Right(options))
+      case _ =>
+        None
+    }
+
+  override def keys = Seq("main-class", "mainClass")
+
+  override def handleValues(
+    values: Seq[Any],
+    cwd: ScopePath,
+    positionOpt: Option[Position]
+  ): Either[BuildException, BuildOptions] = {
+    val mainClasses = DirectiveUtil.stringValues(values)
+    val options = BuildOptions(
+      mainClass = mainClasses.lastOption
```

Comment:
I'm happy to implement your suggestion, but I would like to point out that `SeveralMainClassesFoundError` seems to be intended for the number of main classes _found_ by analyzing the classpath, while this situation is about the number of main classes supplied by directives. That said, supplying multiple --main-class command line arguments also results in an error so this should definitely too, and it seems best to reuse this error rather than be pedantic that it's not about _found_ but _supplied_ main classes.

Reply:
I think too, that is better to reuse `SeveralNoClassdefFoundError` in this case.

Anyway, we started working on an options validation mechanism, which should detect when a directive is specified multiple times, so for now your proposal I good to merge. Later, we will integrate a new mechanism with your directive.